src/plugins/discover/public/application/main/utils/resolve_data_view.ts

```diff
@@ -75,12 +75,28 @@ export function getDataViewId(
  * Function to load the given data view by id, providing a fallback if it doesn't exist
  */
 export async function loadDataView(
-  id: string,
   dataViews: DataViewsContract,
-  config: IUiSettingsClient
+  config: IUiSettingsClient,
+  id?: string
 ): Promise<DataViewData> {
   const dataViewList = (await dataViews.getCache()) as unknown as DataViewSavedObject[];
 
+  try {
+    const fetchedDataView = id ? await dataViews.get(id) : undefined;
```

Comment:
We could simplify it, if `dataViews` would provide something like `dataViews.getAdHocDataViews()`. WDYT @ppisljar?

Reply:
for now we'll try to avoid this, apps should track the adhoc data view on their side, we don't want apps to know about adhoc data views other consumers created.